### PR TITLE
 added support for reporting 

### PR DESCRIPTION
**Pull request by selfagency**
the spec provides for a report uri, however this helmet extension currently does not. i have attempted to address this.

### CODE REVIEW

**Comment by EvanHahn** [comment]
This looks interesting! Could you send me a link to the spec and where it says this?

**Comment by selfagency** [comment]
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection

**Comment by EvanHahn** [comment]
👍, thanks. I'll leave a review now.

File: index.js [hunk 1]
```diff
@@ -1,7 +1,9 @@
 module.exports = function xXssProtection (options) {
+  var checkReport = options && options.reportUri ? '1; report=' + options.reportUri : '1; mode=block'
```
Comment (EvanHahn) [inline index.js]: Nitpick: I think I'd prefer a variable name like `headerValue` for this.

File: README.md [hunk 2]
```diff
@@ -22,3 +22,9 @@ To force the header to be set to `1; mode=block` on all versions of IE, add the
 app.use(xssFilter({ setOnOldIE: true }))
 // This has some security problems for old IE!
 ```
+
+To report errors to a specific URI, add the option:
```
Comment (EvanHahn) [inline README.md]: I'd like to add a link to the part of MDN you specified.

**Comment by EvanHahn** [comment]
Lemme know if you need any help with this—happy to aid.

File: index.js (hunk 1 above)
Comment (selfagency) [reply index.js]: done

File: README.md (hunk 2 above)
Comment (selfagency) [reply README.md]: done

**Comment by selfagency** [comment]
dunno if you got notified, but i implemented all your requested changes

**Comment by EvanHahn** [comment]
This has been released in `x-xss-protection@1.1.0` and `helmet@3.12.0`! Feel free to add yourself to the [list of contributors](https://github.com/helmetjs/helmetjs.github.io/blob/master/contributors/index.md).

To be consistent with existing `mode=block` behavior, I made the header value `1; mode=block; report={{uri}}`. I could see disabling `mode=block` in the future but it feels independent of this change.

For posterity: I made a [demo app to test the `report` directive](https://github.com/EvanHahn/x-xss-protection-report-demo). It worked in Chrome but not in Firefox.

Thanks for your contribution!

**Pull request merged**